File: classes/parte-7-paradigmas/113-oo-basado-en-prototipos-javascript/implementaciones/c/main.c
#include <stdio.h>

struct Obj {
    long valor;
};

long doble(struct Obj *o) {
    return o->valor * 2;
}

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    struct Obj o = {n};
    printf("resultado=%ld\n", doble(&o));
    return 0;
}

File: classes/parte-7-paradigmas/113-oo-basado-en-prototipos-javascript/implementaciones/java/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    static class Obj {
        int valor;
        Obj(int v) { valor = v; }
        int doble() { return valor * 2; }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().trim());
        System.out.println("resultado=" + new Obj(n).doble());
    }
}
